Multisig/Logic/Models/PrivateKey/web3swift/BIP32HDNode.swift

```diff
@@ -0,0 +1,303 @@
+    public static var defaultPath: String = "m/44'/60'/0'/0"
+    public static var defaultPathPrefix: String = "m/44'/60'/0'"
+    public static var defaultPathMetamask: String = "m/44'/60'/0'/0/0"
+    public static var defaultPathMetamaskPrefix: String = "m/44'/60'/0'/0"
+    public static var hardenedIndexPrefix: UInt32 = (UInt32(1) << 31)
+}
+
+extension HDNode {
+    public func derive (index: UInt32, derivePrivateKey:Bool, hardened: Bool = false) -> HDNode? {
+        if derivePrivateKey {
+            if self.hasPrivate { // derive private key when is itself extended private key
+                var entropy:Array<UInt8>
+                var trueIndex: UInt32
+                if index >= (UInt32(1) << 31) || hardened {
+                    trueIndex = index;
+                    if trueIndex < (UInt32(1) << 31) {
+                        trueIndex = trueIndex + (UInt32(1) << 31)
+                    }
+                    let hmac:Authenticator = HMAC(key: self.chaincode.bytes, variant: .sha512)
+                    var inputForHMAC = Data()
+                    inputForHMAC.append(Data([UInt8(0x00)]))
+                    inputForHMAC.append(self.privateKey!)
+                    inputForHMAC.append(trueIndex.serialize32())
+                    guard let ent = try? hmac.authenticate(inputForHMAC.bytes) else {return nil }
+                    guard ent.count == 64 else { return nil }
+                    entropy = ent
+                } else {
+                    trueIndex = index
+                    let hmac:Authenticator = HMAC(key: self.chaincode.bytes, variant: .sha512)
+                    var inputForHMAC = Data()
+                    inputForHMAC.append(self.publicKey)
+                    inputForHMAC.append(trueIndex.serialize32())
+                    guard let ent = try? hmac.authenticate(inputForHMAC.bytes) else {return nil }
+                    guard ent.count == 64 else { return nil }
+                    entropy = ent
+                }
+                let I_L = entropy[0..<32]
+                let I_R = entropy[32..<64]
+                let cc = Data(I_R)
+                let bn = BigUInt(Data(I_L))
+                if bn > HDNode.curveOrder {
+                    if trueIndex < UInt32.max {
+                        return self.derive(index:index+1, derivePrivateKey: derivePrivateKey, hardened:hardened)
+                    }
+                    return nil
+                }
+                let newPK = (bn + BigUInt(self.privateKey!)) % HDNode.curveOrder
+                if newPK == BigUInt(0) {
+                    if trueIndex < UInt32.max {
+                        return self.derive(index:index+1, derivePrivateKey: derivePrivateKey, hardened:hardened)
+                    }
+                    return nil
+                }
+                guard let privKeyCandidate = newPK.serialize().setLengthLeft(32) else {return nil}
```

Comment:
please put space before return, also same for next line

Reply:
will do in the next PR